reconcile/aus/base.py

```diff
@@ -982,6 +983,49 @@ def verify_lock_should_skip(
     return False
 
 
+def verify_max_upgrades_should_skip(
+    desired: ClusterUpgradeSpec,
+    desired_state: OrganizationUpgradeSpec,
+    sector_upgrades: dict[str, set[str]],
+) -> bool:
+    sector_name = desired.upgrade_policy.conditions.sector
+    if not sector_name:
+        return False
+
+    current_upgrades = sector_upgrades[sector_name]
+    # Allow at least one upgrade
+    if len(current_upgrades) == 0:
+        return False
+    sector = desired_state.sectors[sector_name]
+
+    # if sector.max_parallel_upgrades is not set, we allow all upgrades
+    if sector.max_parallel_upgrades is None:
+        return False
+
+    sector_cluster_count = len(sector.specs)
+
+    if sector.max_parallel_upgrades.endswith("%"):
+        max_parallel_upgrades_percent = int(sector.max_parallel_upgrades[:-1])
+        max_parallel_upgrades = round(
+            sector_cluster_count * max_parallel_upgrades_percent / 100
+        )
+    else:
+        max_parallel_upgrades = int(sector.max_parallel_upgrades)
+
+    # we allow at least one upgrade
+    if max_parallel_upgrades == 0:
+        max_parallel_upgrades = 1
```

Comment:
can we limit it at schema level that `max_parallel_upgrades` must be positive int or `1-100%`, like `^(100|[1-9]\d?)%|[1-9]\d*$`. So we don't have special rules about hidden bump it to 1.


```python
        max_parallel_upgrades = math.ceil(
            sector_cluster_count * max_parallel_upgrades_percent / 100
        )
```

Reply:
yes, but if they set 1% on say 50 clusters in the sector, then we will still get `0` here

Reply:
using `ceil` could be invalid in other cases

Reply:
`math.ceil(20 * 1 / 100` = `1`, which case is invalid?